app/controllers/accountants_controller.rb

```diff
@@ -10,10 +10,22 @@ def index
   end
 
   def search
-    @results = if params[:search].present?
-                 Patient.where(pledge_sent: true)
-                        .order_by(pledge_sent_at: :desc)
-                        .search(params[:search])
+    have_search = params[:search].present?
+    have_clinic = params[:clinic_id].present?
+
+    @results = if have_search || have_clinic
+                  partial_result = Patient.where(pledge_sent: true)
+                                          .order_by(pledge_sent_at: :desc)
+
+                  if have_clinic
+                    partial_result = partial_result.where(clinic_id: params[:clinic_id])
+                  end
+
+                  if have_search
+                    partial_result = partial_result.search(params[:search])
+                  end
```

Comment:
This is tickling my *"there must be a more idiomatic way"* spidey sense. Is there a method-call assignment operator? As in `x .= f()` to do `x = x.f()`. If this looks ok I'm fine keeping it though.

Reply:
If I were writing this (ymmv!) I'd probably do something like this, but it's basically the same approach you have going on:

```
result = Patient
result = result.where(clinic..) if params[:clinic_id].present?
result = result.search(...) if params[:search].present?
```
I'm not sure I've ever used a `.=` operator (I lead a somewhat charmed life), so not sure I can offer any advice on that!

Reply:
ok, that's cleaner. thanks!